README.md

```diff
@@ -25,162 +25,90 @@ implementation "com.careem.mockingbird:mockingbird:$mockingBirdVersion"
 
-### Mocks
+### Mock generation plugin ( experimental )
 
-The first step you need to do is create a mock class for the object you want to mock, you need a mock for each
-dependency type you want to mock
+The mock generation plugin generates mock boilerplate code for you, the plugin can be used along with manual mocks,
+it is currently experimental and has several limitations.
 
-The library provides 2 functions to help you write your mocks.
+To use this plugin you have to use mockingbird version `2.0.0-beta04` or above, to see examples checkout `generate-mocks`
+and explore the `samples` folder, `samples` is a project itself, you can open `samples` as a standalone project.
 
-1. `mock` this function allows you to mock non-Unit methods
-1. `mockUnit` this function allows you to mock Unit methods
+NOTE: the plugin doesn't discover which interfaces to mock, it's up to you to configure those.
 
-These helpers enable you to map your mock invocations to MockingBird environment.
+WARNING: If you do not what to use the plugin you can use the old way of manual mock generation
+check [Mocks](https://github.com/careem/mockingbird#mocks) or [Spies](https://github.com/careem/mockingbird#spies)
 
-Your mock class must implement `Mock`, in addition to extending the actual class or implementing an interface
+#### Plugin Setup
 
-See below for an example on how to create a mock :
+To start using the plugin you need to include it in your project `build.gradle.kts` or `build.gradle` adding the
+following line
 
 ```kotlin
-interface MyDependency {
-    fun method1(str: String)
-    fun method2(str: String, value: Int)
-    fun method3(value1: Int, value2: Int): Int
-}
-
-class MyDependencyMock : MyDependency, Mock {
-    object Method {
-        const val method1 = "method1"
-        const val method2 = "method2"
-        const val method3 = "method3"
-        const val method4 = "method4"
-    }
-
-    object Arg {
-        const val str = "str"
-        const val value = "value"
-        const val value1 = "value1"
-        const val value2 = "value2"
-        const val object1 = "object1"
-    }
-
-    override fun method1(str: String) = mockUnit(
-        methodName = Method.method1,
-        arguments = mapOf(
-            Arg.str to str
-        )
-    )
-
-    override fun method2(str: String, value: Int) = mockUnit(
-        methodName = Method.method2,
-        arguments = mapOf(
-            Arg.str to str,
-            Arg.value to value
-        )
-    )
-
-    override fun method3(value1: Int, value2: Int): Int = mock(
-        methodName = Method.method3,
-        arguments = mapOf(
-            Arg.value1 to value1,
-            Arg.value2 to value2
-        )
-    )
-
-    override fun method4(object1: Object): Int = mock(
-        methodName = Method.method4,
-        arguments = mapOf(
-            Arg.object1 to object1
-        )
-    )
-}    
+classpath("com.careem.mockingbird:mockingbird-compiler:2.0.0-beta05")
```

Comment:
```suggestion
classpath("com.careem.mockingbird:mockingbird-compiler:2.0.0-beta06")
```

Reply:
Should we update this version as part of the release pipeline?

Reply:
replaced with variable so we do not need to update